pkg/graph/resolvers/key_contact_category_test.go

```diff
@@ -0,0 +1,421 @@
+
+			updatedCategory, err := UpdateKeyContactCategory(
+				suite.testConfigs.Context,
+				suite.testConfigs.Logger,
+				suite.testConfigs.Principal,
+				suite.testConfigs.Store,
+				category.ID,
+				name,
+			)
+
+			if tc.shouldErr {
+				suite.Error(err)
+				suite.Contains(err.Error(), "category name cannot be empty or whitespace-only")
+			} else {
+				suite.NoError(err)
+				suite.NotNil(updatedCategory)
+				suite.Equal(tc.expected, updatedCategory.Name)
+			}
+		})
+	}
+}
+
+// TestUpdateKeyContactCategory tests updating an existing key contact category.
+func (suite *ResolverSuite) TestUpdateKeyContactCategory() {
+	originalCategoryName := "Original Category"
+	updatedCategoryName := "Updated Category"
+
+	// Create a category first
+	category := suite.createKeyContactCategory(originalCategoryName)
+
+	// Update the category
+	name := updatedCategoryName
+
+	updatedCategory, err := UpdateKeyContactCategory(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		category.ID,
+		name,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(updatedCategory)
+	suite.Equal(category.ID, updatedCategory.ID)
+	suite.Equal(updatedCategoryName, updatedCategory.Name)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, updatedCategory.CreatedBy)
+	suite.NotNil(updatedCategory.ModifiedBy)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, *updatedCategory.ModifiedBy)
+	suite.NotNil(updatedCategory.ModifiedDts)
+}
+
+// TestUpdateKeyContactCategory_NotFound tests updating a non-existent category.
+func (suite *ResolverSuite) TestUpdateKeyContactCategory_NotFound() {
+	nonExistentID := uuid.New()
+	name := "Updated Category"
+
+	_, err := UpdateKeyContactCategory(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nonExistentID,
+		name,
+	)
+
+	suite.Error(err)
+	suite.Contains(err.Error(), "record not found for given key")
+}
+
+// TestDeleteKeyContactCategory tests deleting a key contact category.
+func (suite *ResolverSuite) TestDeleteKeyContactCategory() {
+	category := suite.createKeyContactCategory("Category To Delete")
+
+	// Verify category exists before deletion
+	retrievedCategory, err := GetKeyContactCategory(
+		suite.testConfigs.Context,
+		category.ID,
+	)
+	suite.NoError(err)
+	suite.NotNil(retrievedCategory)
+	suite.Equal(category.ID, retrievedCategory.ID)
+
+	// Delete the category
+	deletedCategory, err := DeleteKeyContactCategory(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		category.ID,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(deletedCategory)
+	suite.Equal(category.ID, deletedCategory.ID)
+
+	// Verify category no longer exists
+	_, err = GetKeyContactCategory(
+		suite.testConfigs.Context,
+		category.ID,
+	)
+	suite.Error(err)
+	suite.Contains(err.Error(), "record not found for given key")
```

Comment:
```suggestion
 suite.True(errors.Is(err, loaders.ErrRecordNotFoundForKey))
```

This is a typed error, so go has functionality to compare it by type instead of by comparing strings.